operations/deployment/terraform/postgres.tf

```diff
@@ -0,0 +1,263 @@
+}
+
+data "aws_vpc" "default" {
+  default = true
+} 
+data "aws_subnets" "vpc_subnets" {
+  filter {
+    name   = "vpc-id"
+
+    # todo: support a specified vpc id
+    # values = [var.vpc_id ? var.vpc_id : data.aws_vpc.default.id]
+    values = [data.aws_vpc.default.id]
+  }
+}
+
+module "rds_cluster" {
+  depends_on     = [data.aws_subnets.vpc_subnets]
+  source         = "terraform-aws-modules/rds-aurora/aws"
+  version        = "v7.6.0"
+  name           = var.aws_resource_identifier
+  engine         = var.postgres_engine
+  engine_version = var.postgres_engine_version
+  instance_class = var.postgres_instance_class
+  instances = {
+    1 = {
+      instance_class = var.postgres_instance_class
+    }
+  }
+
+    # Todo: handle vpc/networking explicitly
+  # vpc_id                 = var.vpc_id
+  # allowed_cidr_blocks    = [var.vpc_cidr]
+  subnets                  = var.postgres_subnets == null || length(var.postgres_subnets) == 0 ? data.aws_subnets.vpc_subnets.ids : var.postgres_subnets
+
+  database_name          = var.postgres_database_name
+  storage_encrypted      = true
+  monitoring_interval    = 60
+  create_db_subnet_group = true
+  db_subnet_group_name   = var.aws_resource_identifier
+  create_security_group  = false
+  vpc_security_group_ids = [aws_security_group.pg_security_group.id]
+
+  # TODO: take advantage of iam database auth
+  iam_database_authentication_enabled    = true
+  master_password                        = random_password.master.result
+  create_random_password                 = false
+  apply_immediately                      = true
+  skip_final_snapshot                    = true
+  create_db_cluster_parameter_group      = true
+  db_cluster_parameter_group_name        = var.aws_resource_identifier
+  db_cluster_parameter_group_family      = "aurora-postgresql11"
+  db_cluster_parameter_group_description = "${var.aws_resource_identifier}  cluster parameter group"
+  db_cluster_parameter_group_parameters = [
+    {
+      name         = "log_min_duration_statement"
+      value        = 4000
+      apply_method = "immediate"
+      }, {
+      name         = "rds.force_ssl"
+      value        = 1
+      apply_method = "immediate"
+    }
+  ]
+  create_db_parameter_group      = true
+  db_parameter_group_name        = var.aws_resource_identifier
+  db_parameter_group_family      = "aurora-postgresql11"
+  db_parameter_group_description = "${var.aws_resource_identifier} example DB parameter group"
+  db_parameter_group_parameters = [
+    {
+      name         = "log_min_duration_statement"
+      value        = 4000
+      apply_method = "immediate"
+    }
+  ]
+  enabled_cloudwatch_logs_exports = ["postgresql"]
+  tags = {
+    Name = "${var.aws_resource_identifier}-RDS"
+  }
+}
+
+resource "random_password" "master" {
+  length = 10
+}
+
+####
+output "postgres_default_subnet_ids" {
+  description = "The subnet ids from the default vpc"
+  value       = data.aws_subnets.vpc_subnets.ids
+}
```

Comment:
this should be moved to the other file/module (removed of the "postgres" reference) along with the data definition above